cdi/embedded/src/test/java/org/infinispan/cdi/embedded/test/distexec/DistributedExecutorCDITest.java

```diff
@@ -62,9 +62,6 @@ public void testInvocationUsingImpliedInputCacheRunnable() throws Exception {
 
    static class SimpleCallable implements Callable<Integer>, Serializable, ExternalPojo {
 
-      /** The serialVersionUID */
-      private static final long serialVersionUID = -8589149500259272402L;
```

Comment:
You actually intrigued me with this... and I think the SO comment is not quite accurate...

So the `Serializable` class [Javadoc](http://docs.oracle.com/javase/7/docs/api/java/io/Serializable.html) states:

> If a serializable class does not explicitly declare a serialVersionUID, then the serialization runtime will calculate a default serialVersionUID value for that class based on various aspects of the class, as described in the Java(TM) Object Serialization Specification. However, it is strongly recommended that all serializable classes explicitly declare serialVersionUID values, since the default serialVersionUID computation is highly sensitive to class details that may vary depending on compiler implementations, and can thus result in unexpected InvalidClassExceptions during deserialization. Therefore, to guarantee a consistent serialVersionUID value across different java compiler implementations, a serializable class must declare an explicit serialVersionUID value.

Now the [Java Serialization Specification](http://docs.oracle.com/javase/8/docs/platform/serialization/spec/class.html#a4100) states:

> The serialVersionUID is computed using the signature of a stream of bytes that reflect the class definition. The National Institute of Standards and Technology (NIST) Secure Hash Algorithm (SHA-1) is used to compute a signature for the stream. The first two 32-bit quantities are used to form a 64-bit hash. A java.lang.DataOutputStream is used to convert primitive data types to a sequence of bytes. The values input to the stream are defined by the Java Virtual Machine (VM) specification for classes. Class modifiers may include the ACC_PUBLIC, ACC_FINAL, ACC_INTERFACE, and ACC_ABSTRACT flags; other flags are ignored and do not affect serialVersionUID computation. Similarly, for field modifiers, only the ACC_PUBLIC, ACC_PRIVATE, ACC_PROTECTED, ACC_STATIC, ACC_FINAL, ACC_VOLATILE, and ACC_TRANSIENT flags are used when computing serialVersionUID values. For constructor and method modifiers, only the ACC_PUBLIC, ACC_PRIVATE, ACC_PROTECTED, ACC_STATIC, ACC_FINAL, ACC_SYNCHRONIZED, ACC_NATIVE, ACC_ABSTRACT and ACC_STRICT flags are used. Names and descriptors are written in the format used by the java.io.DataOutputStream.writeUTF method.

So yes, the `serialVersionUID` is computed in the runtime but based on "the signature of a stream of bytes that reflect the class definition", so based on compiler's output. I believe that's why they say "it is strongly recommended that all serializable classes explicitly declare serialVersionUID values, since the default serialVersionUID computation is highly sensitive to class details that may vary depending on compiler implementations". 

And finally, [this question](http://stackoverflow.com/questions/29081134/why-compiler-do-not-set-serialversionuid-automatically) doesn't make sense I think (assuming that there's no bug in JVM in serialization runtime of course!).